### PR TITLE
Enhancement(CX-1637): Reenable push notifications in user profile setting

### PR DESCRIPTION
**Pull request by lordkiz**
The type of this PR is: **TYPE**

This PR resolves [CX-1637]

### Description

https://user-images.githubusercontent.com/18648835/127615556-46be3372-95fe-447d-b223-33a05c81d20b.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

- Reenable Push Notification item in user profile

#### Dev changes

-

</details>

[CX-1637]: https://artsyproduct.atlassian.net/browse/CX-1637

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ff70ab044cb9fbf6ea972e3877946eeedd3fdc6b
</p>

**Review by pvinis — APPROVED** [review]
I'll approve. Code seems 👌👌, but I have not run this PR on my device to test it.

**Review by brainbicycle — APPROVED** [review]

File: src/lib/Scenes/MyProfile/MyProfilePushNotifications.tsx [hunk 1]
```diff
@@ -7,12 +7,20 @@ import useAppState from "lib/utils/useAppState"
...
 import { MyProfilePushNotifications_me } from "../../../__generated__/MyProfilePushNotifications_me.graphql"
 import { MyProfilePushNotificationsQuery } from "../../../__generated__/MyProfilePushNotificationsQuery.graphql"
 import { updateMyUserProfile } from "../MyAccount/updateMyUserProfile"
 
+const INSTRUCTIONS = {
+  ios: `To receive push notifications from Artsy, you will need to enable them in your iOS Settings. Tap 'Artsy' and
+  toggle "Allow Notifications" on.`,
+  android: `To receive push notifications from Artsy, you will need to enable them in your device settings.
+  In the settings go to "Apps", Tap on 'Artsy' to access Artsy specific settings, go to Notifications and toggle "Show Notifications" on.`,
+}
```
Comment (MounirDhahri) [inline src/lib/Scenes/MyProfile/MyProfilePushNotifications.tsx:22]: **nitpick:** @lordkiz I think we should adjust this content slightly. 
```suggestion
  android: `To receive push notifications from Artsy, you will need to enable them in your device settings.
  Gp to 'Apps', Tap on 'Artsy' to access Artsy specific settings, Tap on 'Notifications' and toggle "Show Notifications" on.`,
}
```

File: src/lib/Scenes/MyProfile/MyProfilePushNotifications.tsx [hunk 2]
```diff
@@ -29,14 +37,16 @@ export const OpenSettingsBanner = () => (
         Artsy would like to send you notifications
       </Sans>
       <Sans size="3t" textAlign="center" color="black60" marginTop="1" marginBottom="2">
-        To receive push notifications from Artsy, you will need to enable them in your iOS Settings. Tap 'Artsy' and
-        toggle "Allow Notifications" on.
+        {Platform.select({ ios: INSTRUCTIONS.ios, android: INSTRUCTIONS.android, default: "" })}
```
Comment (MounirDhahri) [inline src/lib/Scenes/MyProfile/MyProfilePushNotifications.tsx:40]: **nitpick:** you can return here `INSTRUCTIONS` and do the `Platform.select` check already when you are creating instructions in line 17 ;).

File: src/lib/Scenes/MyProfile/MyProfilePushNotifications.tsx [hunk 3]
```diff
@@ -102,20 +112,26 @@ export const MyProfilePushNotifications: React.FC<{
   const [userNotificationSettings, setUserNotificationSettings] = useState<MyProfilePushNotifications_me>(me)
   const [isRefreshing, setIsRefreshing] = useState<boolean>(false)
 
-  useEffect(() => {
-    LegacyNativeModules.ARTemporaryAPIModule.fetchNotificationPermissions((_, result: PushAuthorizationStatus) => {
-      setNotificationAuthorizationStatus(result)
-    })
-  }, [])
+  useEffect(() => getPermissionStatus(), [])
```
Comment (MounirDhahri) [inline src/lib/Scenes/MyProfile/MyProfilePushNotifications.tsx:115]: **thought:** I think you want this instead right?
```suggestion
  useEffect(() => {
      getPermissionStatus()
  }, [])
```

File: src/lib/Scenes/MyProfile/MyProfile.tsx [hunk 4]
```diff
@@ -34,6 +37,18 @@ const MyProfile: React.FC<{ me: MyProfile_me; relay: RelayRefetchProp }> = ({ me
...
+  useEffect(() => {
+    AsyncStorage.getItem(ASYNC_STORAGE_PUSH_NOTIFICATIONS_KEY).then((canDisplay) => {
+      setShouldDisplayPushNotifications(
+        Platform.select({
+          ios: true,
+          android: canDisplay === "true",
+          default: shouldDisplayPushNotifications,
+        })
+      )
+    })
```
Comment (MounirDhahri) [inline src/lib/Scenes/MyProfile/MyProfile.tsx:49]: **issue(blocker):** setting `shouldDisplayPushNotifications` to the same value on iOS (true), will unfortunately trigger a re-render that can be avoid and that's due to the way react works. I guess you can avoid that by simply doing as follows: 
```suggestion
if (Platform.OS === "android") {
    AsyncStorage.getItem(ASYNC_STORAGE_PUSH_NOTIFICATIONS_KEY).then((canDisplay) => {
      setShouldDisplayPushNotifications(canDisplay === "true")
    })
  }
```

**Review by MounirDhahri — COMMENTED** [review]
This is looking great, I left few minor comments and we should be clear to ⛴⛴⛴

File: src/lib/Scenes/MyProfile/MyProfilePushNotifications.tsx (hunk 3 above)
Comment (lordkiz) [reply src/lib/Scenes/MyProfile/MyProfilePushNotifications.tsx:115]: Oh my! Yes. 🤦

File: src/lib/Scenes/MyProfile/MyProfile.tsx (hunk 4 above)
Comment (lordkiz) [reply src/lib/Scenes/MyProfile/MyProfile.tsx:49]: Oh I had always thought that rerenders are not triggered when using useState in functional components if the incoming value passes the === comparison.
Changing it to be safe. Thanks.

**Review by MounirDhahri — APPROVED** [review]

**Comment by MounirDhahri** [comment]
Niiiice ⛴⛴⛴

**Pull request merged**